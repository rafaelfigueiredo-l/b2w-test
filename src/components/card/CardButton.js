import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const CardButton = (props) => {

  const {getRandomPlanet} = props;

  return (
    <>
      <Wrapper><Button onClick={() => getRandomPlanet()}>NEXT</Button></Wrapper>
    </>
  );
};

const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid #FFE919;
  color: #FFE919;
  margin-top: 20px;
  padding: 20px;
  cursor: pointer;
  font-size: 22px;
  text-align: center;
`;
const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

CardButton.propTypes = {
  getRandomPlanet: PropTypes.func.isRequired,
};

export default CardButton;
