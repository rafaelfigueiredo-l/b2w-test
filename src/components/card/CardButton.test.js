import CardButton from './CardButton';
const func = () => {
    return true;
};

describe('<CardButton />', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<CardButton getRandomPlanet={func} />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Snapshot testing', () => {
      const wrapper = shallow(<CardButton getRandomPlanet={func}  />);
      expect(wrapper).toMatchSnapshot();
    });
});