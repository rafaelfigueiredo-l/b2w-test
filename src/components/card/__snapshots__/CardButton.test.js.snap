// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardButton /> Snapshot testing 1`] = `
<Fragment>
  <styled.div>
    <styled.button
      onClick={[Function]}
    >
      NEXT
    </styled.button>
  </styled.div>
</Fragment>
`;
